src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -393,6 +400,25 @@ private void ClaimPeerTip(int networkPeerId, uint256 tipHash)
             peersClaimingThisHeader.Add(networkPeerId);
         }
 
+        /// <inheritdoc />
+        public void ClaimPeerTipForMiner(uint256 tipHash)
+        {
+            HashSet<int> peersClaimingThisHeader;
+            if (!this.peerIdsByTipHash.TryGetValue(tipHash, out peersClaimingThisHeader))
+            {
+                this.peerIdsByTipHash.Add(tipHash, new HashSet<int>(new[] { MinerPeerId }));
+            }
+            else
+            {
+                // Update the last miner tip hash with the new tip
+                var lastMinerTip = this.peerIdsByTipHash.FirstOrDefault(p => p.Value.Contains(MinerPeerId));
```

Comment:
This is probably wrong we need to walk back the current branch otherwise we risk removing mined blocks from parallel branches

Reply:
Mining will always happen at the tip so how can parallel branches with the miner peer id happen? So every time we mine a block we are at tip so we need to update the last mined tip to the new one?

If my understanding is correct then "branches" refers to tips by peer id, so I'm not how a miner in this case can claim more than one tip? Seeing as its always at tip.

Just thinking out loud here trying to understand this better :)

Reply:
Maybe give me a scenario of where this will be possible?

Reply:
yeah basically we can have more then one branch, but the miner will always extend the longest validated branch, however after a reorg we do not want to delete blocks that we mined on a shorter branch.
In the case bellow if we mined 6A, and then a reorg happened and we moved to chain B and then we mined block 8B we should have 6A marked as -2 and 8B, so you dont want to delete -2 from 6A
```
1A-2A-3A-4A-5A-6A
                -4B-5B-6B-7B-8B-9B
```